### ISSUE TITLE
IPSEC DOCKER MONITORING

### ISSUE THREAD

**Issue by Gero6700** [akpw/mktxp-stack#26]
Could you tell me or add that in prometheus get the IPSEC data? 

I have checked the changes you made in the mktxp without dockerizing and it is correct but it doesn't get the data.

I put you the image and the changes inside the container

![imagen](https://github.com/user-attachments/assets/97c1bb57-6a34-4291-9c68-d9bcc2b7e5d1)

``` sh
gdonaire@HomeLab:~/mktxp-stack$ grep -r "ipsec" .
./mktxp/mktxp.conf:    ipsec = True
./grafana/provisioning/dashboards/mikrotik/mikrotik_loki_logs.json:            "text": "ipsec",
./grafana/provisioning/dashboards/mikrotik/mikrotik_loki_logs.json:            "value": "ipsec"
./grafana/provisioning/dashboards/mikrotik/mikrotik_loki_logs.json:        "query": "account,caps,critical,error,firewall,info,warning,wireless,dns,health,dhcp,interface,ipsec,l2tp,ppp,raw,ssh,system,ups",
gdonaire@HomeLab:~/mktxp-stack$ ls
LICENSE  README.md  docker-compose-mktxp-stack-no-logs.yml  docker-compose-mktxp-stack.yml  docker-compose-mktxp-with-logs.yml  grafana  loki  mktxp  prometheus  promtail  syslog-ng
gdonaire@HomeLab:~/mktxp-stack$ docker exec -it mktxp sh
/mktxp # ls
LICENSE         README.md       build           deploy          mktxp           mktxp.egg-info  setup.py
/mktxp # grep -r "ipsec" .
./build/lib/mktxp/datasource/ipsec_ds.py:            ipsec_records = router_entry.api_connection.router_api().get_resource('/ip/ipsec/active-peers').call('print', {'stats': ''})
./build/lib/mktxp/datasource/ipsec_ds.py:            for record in ipsec_records:
./build/lib/mktxp/datasource/ipsec_ds.py:            return BaseDSProcessor.trimmed_records(router_entry, router_records=ipsec_records,
./build/lib/mktxp/cli/config/config.py:    FE_IPSEC_KEY = 'ipsec'
./build/lib/mktxp/cli/config/mktxp.conf:    ipsec = False                   # IPSec active peer metrics
./build/lib/mktxp/flow/collector_registry.py:from mktxp.collector.ipsec_collector import IPSecCollector
./build/lib/mktxp/collector/ipsec_collector.py:from mktxp.datasource.ipsec_ds import IPSecMetricsDataSource
./build/lib/mktxp/collector/ipsec_collector.py:        if not router_entry.config_entry.ipsec:
./build/lib/mktxp/collector/ipsec_collector.py:        ipsec_labels = ['local_address', 'remote_address', 'name', 'last_seen', 'uptime', 'ph2_total', 'responder',
./build/lib/mktxp/collector/ipsec_collector.py:        ipsec_records = IPSecMetricsDataSource.metric_records(router_entry, metric_labels=ipsec_labels,
./build/lib/mktxp/collector/ipsec_collector.py:        if ipsec_records:
./build/lib/mktxp/collector/ipsec_collector.py:            ipsec_info_labels = ['local_address', 'name', 'remote_address', 'state']
./build/lib/mktxp/collector/ipsec_collector.py:            ipsec_state_metric = BaseCollector.info_collector('ipsec_peer_state',
./build/lib/mktxp/collector/ipsec_collector.py:                                                              ipsec_records, ipsec_info_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            yield ipsec_state_metric
./build/lib/mktxp/collector/ipsec_collector.py:            ipsec_value_labels = ['local_address', 'name', 'remote_address']
./build/lib/mktxp/collector/ipsec_collector.py:            rx_byte_metric = BaseCollector.counter_collector('ipsec_peer_rx_byte',
./build/lib/mktxp/collector/ipsec_collector.py:                                                             ipsec_records, 'rx_bytes', ipsec_value_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            tx_byte_metric = BaseCollector.counter_collector('ipsec_peer_tx_byte',
./build/lib/mktxp/collector/ipsec_collector.py:                                                             ipsec_records, 'tx_bytes', ipsec_value_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            rx_packet_metric = BaseCollector.counter_collector('ipsec_peer_rx_packet',
./build/lib/mktxp/collector/ipsec_collector.py:                                                               ipsec_records, 'rx_packets', ipsec_value_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            tx_packet_metric = BaseCollector.counter_collector('ipsec_peer_tx_packet',
./build/lib/mktxp/collector/ipsec_collector.py:                                                               ipsec_records, 'tx_packets', ipsec_value_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            p2_total_metric = BaseCollector.gauge_collector('ipsec_peer_security_association',
./build/lib/mktxp/collector/ipsec_collector.py:                                                             ipsec_records, 'p2_total', ipsec_value_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            last_seen_metric = BaseCollector.gauge_collector('ipsec_peer_last_seen',
./build/lib/mktxp/collector/ipsec_collector.py:                                                             ipsec_records, 'last_seen', ipsec_value_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            uptime_metric = BaseCollector.gauge_collector('ipsec_peer_uptime', 'How long peer is in an established state.',
./build/lib/mktxp/collector/ipsec_collector.py:                                                          ipsec_records, 'uptime', ipsec_value_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            responder_metric = BaseCollector.gauge_collector('ipsec_peer_responder',
./build/lib/mktxp/collector/ipsec_collector.py:                                                          ipsec_records, 'responder', ipsec_value_labels)
./build/lib/mktxp/collector/ipsec_collector.py:            natt_metric = BaseCollector.gauge_collector('ipsec_peer_natt_enabled',
./build/lib/mktxp/collector/ipsec_collector.py:                                                          ipsec_records, 'natt_peer', ipsec_value_labels)
./mktxp.egg-info/PKG-INFO:    ipsec = False                   # IPSec active peer metrics
./mktxp.egg-info/SOURCES.txt:mktxp/collector/ipsec_collector.py
./mktxp.egg-info/SOURCES.txt:mktxp/datasource/ipsec_ds.py
./mktxp/datasource/ipsec_ds.py:            ipsec_records = router_entry.api_connection.router_api().get_resource('/ip/ipsec/active-peers').call('print', {'stats': ''})
./mktxp/datasource/ipsec_ds.py:            for record in ipsec_records:
./mktxp/datasource/ipsec_ds.py:            return BaseDSProcessor.trimmed_records(router_entry, router_records=ipsec_records,
./mktxp/cli/config/config.py:    FE_IPSEC_KEY = 'ipsec'
./mktxp/cli/config/mktxp.conf:    ipsec = False                   # IPSec active peer metrics
./mktxp/flow/collector_registry.py:from mktxp.collector.ipsec_collector import IPSecCollector
./mktxp/collector/ipsec_collector.py:from mktxp.datasource.ipsec_ds import IPSecMetricsDataSource
./mktxp/collector/ipsec_collector.py:        if not router_entry.config_entry.ipsec:
./mktxp/collector/ipsec_collector.py:        ipsec_labels = ['local_address', 'remote_address', 'name', 'last_seen', 'uptime', 'ph2_total', 'responder',
./mktxp/collector/ipsec_collector.py:        ipsec_records = IPSecMetricsDataSource.metric_records(router_entry, metric_labels=ipsec_labels,
./mktxp/collector/ipsec_collector.py:        if ipsec_records:
./mktxp/collector/ipsec_collector.py:            ipsec_info_labels = ['local_address', 'name', 'remote_address', 'state']
./mktxp/collector/ipsec_collector.py:            ipsec_state_metric = BaseCollector.info_collector('ipsec_peer_state',
./mktxp/collector/ipsec_collector.py:                                                              ipsec_records, ipsec_info_labels)
./mktxp/collector/ipsec_collector.py:            yield ipsec_state_metric
./mktxp/collector/ipsec_collector.py:            ipsec_value_labels = ['local_address', 'name', 'remote_address']
./mktxp/collector/ipsec_collector.py:            rx_byte_metric = BaseCollector.counter_collector('ipsec_peer_rx_byte',
./mktxp/collector/ipsec_collector.py:                                                             ipsec_records, 'rx_bytes', ipsec_value_labels)
./mktxp/collector/ipsec_collector.py:            tx_byte_metric = BaseCollector.counter_collector('ipsec_peer_tx_byte',
./mktxp/collector/ipsec_collector.py:                                                             ipsec_records, 'tx_bytes', ipsec_value_labels)
./mktxp/collector/ipsec_collector.py:            rx_packet_metric = BaseCollector.counter_collector('ipsec_peer_rx_packet',
./mktxp/collector/ipsec_collector.py:                                                               ipsec_records, 'rx_packets', ipsec_value_labels)
./mktxp/collector/ipsec_collector.py:            tx_packet_metric = BaseCollector.counter_collector('ipsec_peer_tx_packet',
./mktxp/collector/ipsec_collector.py:                                                               ipsec_records, 'tx_packets', ipsec_value_labels)
./mktxp/collector/ipsec_collector.py:            p2_total_metric = BaseCollector.gauge_collector('ipsec_peer_security_association',
./mktxp/collector/ipsec_collector.py:                                                             ipsec_records, 'p2_total', ipsec_value_labels)
./mktxp/collector/ipsec_collector.py:            last_seen_metric = BaseCollector.gauge_collector('ipsec_peer_last_seen',
./mktxp/collector/ipsec_collector.py:                                                             ipsec_records, 'last_seen', ipsec_value_labels)
./mktxp/collector/ipsec_collector.py:            uptime_metric = BaseCollector.gauge_collector('ipsec_peer_uptime', 'How long peer is in an established state.',
./mktxp/collector/ipsec_collector.py:                                                          ipsec_records, 'uptime', ipsec_value_labels)
./mktxp/collector/ipsec_collector.py:            responder_metric = BaseCollector.gauge_collector('ipsec_peer_responder',
./mktxp/collector/ipsec_collector.py:                                                          ipsec_records, 'responder', ipsec_value_labels)
./mktxp/collector/ipsec_collector.py:            natt_metric = BaseCollector.gauge_collector('ipsec_peer_natt_enabled',
./mktxp/collector/ipsec_collector.py:                                                          ipsec_records, 'natt_peer', ipsec_value_labels)
./README.md:    ipsec = False                   # IPSec active peer metrics
./deploy/kubernetes/secret.yaml:            ipsec = False                   # IPSec active peer metrics`

**Comment by akpw** [comment]
@Gero6700 can you run & share the results of `$ mktxp show`? 
you can also check the mktxp.conf used with `$ mktxp show -cfg`, to make sure the right config file is being used

adding @phibos here as well

**Comment by Gero6700** [comment]
If of course, whatever you ask me, to solve it and to learn how it works a little bit too.

```bash

/mktxp # mktxp show
[Router-Home]
    enabled: True
    hostname: 10.12.0.5
    port: 8728

    username: mktxp_monitoring
    password: *************

    use_ssl: False
    no_ssl_certificate: False
    ssl_certificate_verify: False
    plaintext_login: True

    dhcp: True
    installed_packages: True
    dhcp_lease: True
    interface: True
    monitor: True
    wireless: True
    wireless_clients: True
    connections: True
    connection_stats: False
    capsman: True
    capsman_clients: True
    poe: True
    netwatch: True
    use_comments_over_names: True
    public_ip: True
    route: True
    pool: True
    firewall: True
    neighbor: True
    ipv6_route: False
    ipv6_pool: False
    ipv6_firewall: False
    ipv6_neighbor: False
    user: True
    queue: True
    remote_dhcp_entry: None
    remote_capsman_entry: None
    check_for_updates: False
    bgp: False
    kid_control_assigned: False
    kid_control_dynamic: False
    lte: False
    ipsec: True
```

``` bash

/mktxp # mktxp show -cfg
MKTXP data config: /root/mktxp/mktxp.conf
MKTXP internal config: /root/mktxp/_mktxp.conf
```

**Comment by akpw** [comment]
OK the config looks good. @phibos, can you check on that one?

**Comment by phibos** [comment]
- What RouterOS version are you using?
- Do you get any entries if you run ```/ip/ipsec/active-peers/print``` on the device?
- Do you get any ipsec output if you access the exporter directly via http?

**Comment by Gero6700** [comment]
- What RouterOS version are you using?
7.15.1
- Do you get any entries if you run /ip/ipsec/active-peers/print on the device?
![image](https://github.com/user-attachments/assets/04754131-2201-4b49-bd20-3848d610f258)

- Do you get any ipsec output if you access the exporter directly via http?
I guess you mean this and it doesn't come out...
![image](https://github.com/user-attachments/assets/5e6d3c7a-6ba6-4dc5-948b-a0b270e702e0)

**Comment by phibos** [comment]
I'm sorry but I'm unable to reproduce the error.

I have copied the config from https://github.com/akpw/mktxp-stack/issues/26#issuecomment-2254552096

Tested with RouterOS
- 7.14.2
- 7.15.2

Tested with environment
- Python 3.12 with venv
- Docker Image ghcr.io/akpw/mktxp:gha-10158298014 from today

Prometheus (docker)
- 2.51.0
- 2.53.1

And I get the metrics.

> * Do you get any ipsec output if you access the exporter directly via http?

Use wget or a browser to access ```<address of exporter>:49090``` and search for ipsec.
